File: service-core/internal/modules/address/domain/address.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type Address struct {
	ID uuid.UUID

	UserID uuid.UUID

	ReceiverName string
	Phone        *string

	IsDefault bool

	ProvinceID  string
	CityID      string
	DistrictID  string
	VillageID   string
	FullAddress string
	PostalCode  string

	CreatedAt time.Time
	UpdatedAt *time.Time
	DeletedAt *time.Time
}

File: e-commerce/app/types/address.ts
// app/types/address.ts

export interface UserAddress {
  address_id?: string
  user_id?: string
  receiver_name: string
  phone: string
  is_default: boolean
  province_id: string
  city_id: string
  district_id: string
  village_id: string
  full_address: string
  postal_code: string
  created_at?: string
}

export interface SaveAddressPayload {
  address_id?: string
  receiver_name: string
  phone: string
  is_default: string // "True" or "False"
  province_id: string
  city_id: string
  district_id: string
  village_id: string
  full_address: string
  postal_code: string
}

export interface ListAddressesResponse {
  addresses: UserAddress[]
}

export interface LocationItem {
  id: string
  name: string
}
